app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/landing/LandingViewModel.kt

```diff
@@ -198,6 +209,17 @@ class LandingViewModel @Inject constructor(
             )
         }
     }
+
+    /**
+     * If the user state account is changed to an active but not "logged in" account we can
+     * pre-populate the email field with this account.
+     */
+    private fun handleActiveAccountChange(activeAccount: UserState.Account) {
+        val activeUserNotLoggedIn = activeAccount.isLoggedIn.not()
+        if (activeUserNotLoggedIn) {
```

Comment:
From my understanding of how we are currently handling that is separate from this, basically if you have that toggled on type in something and continue that would be saved, and the next time you landed on the `LandingScreen` it would repopulate with that. This would supersede that in the event the user switched to the "soft" logged out event. One side effect (potential bug) I will call out is that if you do this, get to the login page/point where you can "add new account" if you do this while the soft logged out user is the active user...then it will autofill that, this IS the same in the current play store app, but maybe there is a way to have some hierarchy.

<img src="https://media3.giphy.com/media/l0IylOPCNkiqOgMyA/giphy.gif"/>

Reply:
```kotlin
private fun handleActiveAccountChange(activeAccount: UserState.Account) {
        val activeUserNotLoggedIn = activeAccount.isLoggedIn.not()
        val noPendingAdditions = authRepository.hasPendingAccountAddition.not()
        if (activeUserNotLoggedIn && noPendingAdditions) {
            trySendAction(LandingAction.EmailInputChanged(activeAccount.email))
        }
    }
```
this would fix that case I believe

Reply:
Uploading yep.mp4…

Reply:
https://github.com/user-attachments/assets/e6b82c85-15f8-4e1f-b1f8-35465d851167